build.gradle

```diff
@@ -185,16 +183,104 @@ clean.doLast {
     println "Cleaned root - don't forget to re-extract stuff! 'gradlew extractNatives extractConfig' will do so, or 'gradlew idea' (or eclipse)"
 }
 
-task protobufCompileWindows(type:Exec) {
+task protobufCompileWindows(type: Exec) {
     description = "Run 'Protobuf Compiler' (Windows)"
     commandLine 'protobuf\\compiler\\protoc.exe', '--proto_path=engine\\src\\main\\protobuf', '--java_out', 'engine\\src\\main\\java', 'engine\\src\\main\\protobuf\\*'
 }
 
-task protobufCompileLinux(type:Exec) {
+task protobufCompileLinux(type: Exec) {
     description = "Run 'Protobuf Compiler' (Linux)"
     commandLine 'protobuf/compiler/protoc', '--proto_path=engine/src/main/protobuf', '--java_out', 'engine/src/main/java', "engine/src/main/protobuf/EntityData.proto", "engine/src/main/protobuf/NetMessage.proto"
 }
 
+
+
+// Magic for replace remote dependency on local project (source)
+// for Engine
+allprojects {
+    configurations.all {
+        resolutionStrategy.dependencySubstitution {
+            substitute module("org.terasology.engine:engine") because "we have sources!" with project(":engine")
+            substitute module("org.terasology.engine:engine-tests") because "we have sources!" with project(":engine-tests")
+        }
+    }
+}
+
+// Magic for replace remote dependency on local project (source)
+// For exists modules
+project(":modules").subprojects.forEach { proj ->
```

Comment:
This block appears to be what can cause a stall with `gradlew idea` when `org.gradle.parallel=true` is set in `gradle.properties` - FYI @DarkWeird 

If I comment this block out then the stall doesn't happen (but I'd assume module dependencies won't work). Probably relates to multiple threads working different modules for the `ideaModule` task yet this block iterates through all modules and for each goes and looks for every other module and substitutes stuff? That seems a classic case for `ConcurrentModificationException` being thrown when working with a `Map` just to come up with a possibly similar example :-)

We could still merge and advise to keep parallel processing off, with the intent to later get rid of `gradlew idea` entirely - *but* there could be a possibility of similar stalls in other Gradle logic that needs to iterate over modules?

Running Gradle in parallel mode is nice for efficiency but probably not really high priority yet. Could be more of an optimization thing to review later :-)